content/blog/2016/2016-04-07-pipeline-for-runs-on-hardware.adoc

```diff
@@ -0,0 +1,258 @@
+(link:https://issues.jenkins-ci.org/browse/JENKINS-30269[JENKINS-30269]).
+Another case is integration with 
+link:https://wiki.jenkins-ci.org/display/JENKINS/Throttle+Concurrent+Builds+Plugin[Throttle Concurrent Builds plugin], which is an effective engine for limiting the license utilization in automation infrastructures 
+(link:https://issues.jenkins-ci.org/browse/JENKINS-31801[JENKINS-31801]).
+* *Better support of CLI tools*. 
+EDA tools frequently need a complex environment, which should be deployed on nodes somehow. 
+Integration with 
+link:https://wiki.jenkins-ci.org/display/JENKINS/Custom+Tools+Plugin[Custom Tools Plugin] seems to be the best option, especially in the case of multiple tool versions 
+(link:https://issues.jenkins-ci.org/browse/JENKINS-30680[JENKINS-30680]).
+* *Pipeline package manager* with dependency management, which would allow developing Pipeline libraries and sharing them between teams. 
+link:https://github.com/jenkinsci/workflow-plugin/blob/master/cps-global-lib/README.md[Pipeline Global Library] and
+link:https://github.com/jenkinsci/workflow-remote-loader-plugin[Pipeline Remote Loader] can be used as a workaround.
+* *Pipeline debugger*. HW test runs are very slow, so it is difficult to troubleshoot and fix issues in the Pipeline code if you have to run every build from scratch. 
+There are several features in Pipeline, which simplify the development, but we still need a full-featured IDE.
+
+## Conclusions
+
+Jenkins is a powerful *automation framework*, which can be used in many areas.
+Even though Jenkins has no dedicated plugins for test runs on hardware, it provides many general-purpose "building blocks", which allow implementing almost any flow.
+That's why Jenkins is so popular in the hardware and embedded areas.
+
+Pipeline-as-code is an emerging technology, which should greatly simplify the implementation of complex flows.
+Currently it lacks integrations with particular Jenkins features, but hopefully this issue will be solved soon.
+
+If you develop new automation flows, consider Pipeline as one of possible approaches.
+
+## What's next?
+
+Jenkins automation server dominates in the HW/embedded area, but unfortunately there is not so much experience sharing for these use-cases. 
+So Jenkins community encourages everybody to share the experience in this area by writing docs and articles for Jenkins website and other resources.
+
+This is just a a first blog post on this topic.
+I am planning to provide more examples of Pipeline-as-code usage for Embedded and Hardware tests in the future posts.
+The next post will be about concurrency and shared resource management in Pipelines.
+
+I am also going to talk about running tests on hardware at the 
+link:https://www.eventbrite.com/e/accelerating-automotive-innovation-with-continuous-integration-delivery-tickets-20809772590[incoming Automotive event] in Stuttgart on April 26th.
+This event is being held by 
```

Comment:
Whoops, I screwed up my comment. I meant to say: "s/held/hosted/"

That said, I think this is a stylistic nitpick. I see people use the verb "hold" in English frequently in contexts similar to this despite "host" being a clearer verb. (seriously, [hold](http://www.thefreedictionary.com/hold) has a *dozen* meanings! what a silly language!)